architect/systems/spectrometers.py

```diff
@@ -221,20 +221,22 @@ def get_spatial_resolution(self, wavelength, target_distance, skew_angle=0):
         Ref: https://www.notion.so/utat-ss/Absolute-Spatial-Resolution-bd475362664e46578b113ff3bfb51e76
 
         """
-
-        sensor_spatial_resolution = self.get_sensor_spatial_resolution(
-            target_distance=target_distance
-        )
-
-        optical_spatial_resolution = self.get_optical_spatial_resolution(
-            wavelength=wavelength,
-            target_distance=target_distance,
-            skew_angle=skew_angle,
-        )
-
-        spatial_resolution = np.maximum(
-            sensor_spatial_resolution, optical_spatial_resolution
-        )
+        if self.spatial_resolution is None:
+            sensor_spatial_resolution = self.get_sensor_spatial_resolution(
+                target_distance=target_distance
+            )
+
+            optical_spatial_resolution = self.get_optical_spatial_resolution(
+                wavelength=wavelength,
+                target_distance=target_distance,
+                skew_angle=skew_angle,
+            )
+
+            spatial_resolution = np.maximum(
+                sensor_spatial_resolution, optical_spatial_resolution
+            )
+        else:
+            spatial_resolution = self.spatial_resolution
```

Comment:
Let's restructure this conditional as follows for clarity
```suggestion
        if self.spatial_resolution is not None:
            spatial_resolution = self.spatial_resolution
        else:
            sensor_spatial_resolution = self.get_sensor_spatial_resolution(
                target_distance=target_distance
            )

            optical_spatial_resolution = self.get_optical_spatial_resolution(
                wavelength=wavelength,
                target_distance=target_distance,
                skew_angle=skew_angle,
            )

            spatial_resolution = np.maximum(
                sensor_spatial_resolution, optical_spatial_resolution
            )
```